Feature/hub 26 cigarettes per day and in pack screens

VikkiRy opened the pull request:
**Экран ввода данных (количество употребляемых сигарет в день)** 

- По открытию на экране ввода данных количества употребляемых сигарет в день отображено поле для ввода количества сигарет в день, неактивный элемент для перехода на следующий экран и активная стрелка для возврата на предыдущий экран.
- Элемент для перехода на следующий экран становиться активны при заполнении поля на экране.
- Поле является заполненным, когда пользователь ввел как минимум один символ.
- Пользователь может ввести данные в поле данных.
- Поле для ввода количества сигарет в день обязательно для заполнения.
- В поле есть подсказка “Введите количество потребляемых сигарет в день” о вводе данных. Подсказка исчезает, когда пользователь нажимает на поле.
- Для ввода данных в поле используются только целые цифровые значения (от 0 до 9). Минимальное числовое значение для ввода - 1, максимальное - 9999.
- При нажатии активного элемента для перехода на следующий экран при использовании числовой значение вне диапазона максимального и минимального числового значения (от 1 до 9999) система отображает на экране текст ошибки “Минимальное числовое значение - 1, максимальное - 9999”, поле загорается красным и кнопка для перехода на следующий экран становится неактивной.
- Текст ошибки исчезает, когда пользователь нажимает на поле для ввода данных.
- Возврат на предыдущий экран (экран ввода имени) осуществляется активной стрелкой.
- После заполнения данных количества употребляемых сигарет в день пользователь может перейти на следующий экран ввода данных через активный элемент перехода на следующий экран.

https://github.com/The-Lessoner/bad_habbit/assets/97001077/c5aaad82-d1a9-4be1-ac1d-e2b1c156a664

VikkiRy committed:
- fastfile was refactored
- BadHabbitDebug.entitlements was removed
- DEVELOPMENT_TEAM was set to none
- PRODUCT_BUNDLE_IDENTIFIER was hanged to com.theLessoner.BadHabit

VikkiRy committed:
.gitignore and BadHabbit.entitlements were changed

VikkiRy committed:
- BadHabbitDebug.entitlenents was removed from project folder
- cigarettesPerDayScreen was added
- Colors asset, LayoutConstants, Localizable.strings were updated
- sf-pro-display-regular font and TextField class was added

VikkiRy committed:
- CigarettesInPackScreen was added
- Localizable.strings was changed
- SignUpViewController, StoriesAssembly were updated

VikkiRy commented:
**Экран ввода данных (количество сигарет в пачке)**

- По открытию на экране ввода данных количества сигарет в пачке отображено поле для ввода количества сигарет в пачке, неактивный элемент для перехода на следующий экран и активная стрелка для возврата на предыдущий экран.  
- Элемент для перехода на следующий экран становиться активны при заполнении поля на экране. 
- Поле является заполненным, когда пользователь ввел как минимум один символ.
- Пользователь может ввести данные в поле данных.
- Поле для ввода количества сигарет в пачке обязательно для заполнения. 
- В поле есть подсказка “Введите количество сигарет в пачке” о вводе данных. Подсказка появляется при попадании пользователем на экран. При вводе данных в поле подсказка исчезает. 
- Для ввода данных в поле используются только целые цифровые значения (от 0 до 9). Минимальное числовое значение для ввода - 1, максимальное - 9999. 
- При нажатии активного элемента для перехода на следующий экран при использовании числовой значение вне диапазона максимального и минимального числового значения (от 1 до 9999) система отображает на экране текст ошибки “Минимальное числовое значение - 1, максимальное - 9999”, поле загорается красным и кнопка для перехода на следующий экран становится неактивной.
- Текст ошибки исчезает, когда пользователь нажимает на поле для ввода данных.
- Возврат на предыдущий экран (экран ввода кол-во употр. сигарет в день) осуществляется  активной стрелкой.
- После заполнения данных количества сигарет в пачке пользователь может перейти на следующий экран ввода данных через активный элемент перехода на следующий экран.


https://github.com/The-Lessoner/bad_habbit/assets/97001077/dbfdef64-627e-4e99-ac99-7ba15c9be3ff

VikkiRy commented:
@nikGayko Do we need to save user input data in a coredata in this pr?

nikGayko commented:
> @nikGayko Do we need to save user input data in a coredata in this pr?

Yes, let's save into core data. Also check comment in Halina MR, there are some basic concept for input view

VikkiRy committed:
Merge branch 'develop' into feature/HUB-26-cigarettes-per-day-and-in-pack-screens

# Conflicts:
#	BadHabbit/BadHabbit.xcodeproj/project.pbxproj
#	BadHabbit/BadHabbit/UserStories/BreakHabitScreen/BreakHabitScreenAssembly.swift
#	BadHabbit/BadHabbit/UserStories/BreakHabitScreen/BreakHabitScreenPresenter.swift
#	BadHabbit/BadHabbit/UserStories/BreakHabitScreen/BreakHabitScreenRouter.swift
#	BadHabbit/BadHabbit/UserStories/DI/StoriesAssembly.swift

VikkiRy committed:
- isValidCigarettesCount func, CigarettesInPackViewController, CigarettesPerDayViewController, TextField were refactored
- emptyLabelPromptText value was added to Localizable.string

VikkiRy committed:
- PersistentKitAssembly, CoreData entities were created
- CigarettesPerDay and CigarettesInPack viewControllers and presenters were refactored
- TextField class was refactored

VikkiRy commented:
https://github.com/The-Lessoner/bad_habbit/assets/97001077/7b7972e7-c43e-4138-a5e3-d968d0631afe

VikkiRy committed:
cigarettesInPackScreen and cigarettesPerDayScreen titleText, cigarettesCountPromptText were changed

VikkiRy committed:
Added text formatting to the textField in CigarettesInPackViewController and CigarettesPerDayViewController
